README.md

```diff
@@ -1,76 +1,124 @@
 # Jenkins Pipeline
 
-Jenkins Pipeline Shared Library, contains helper functions to be used with the Jenkins Pipeline Plugin.
+Jenkins Pipeline Shared Library.  Contains helper functions to be used with the Jenkins Pipeline Plugin.
 
-## Configuration
+## Image Class
 
-The `Docker` library assumes a `Map` of specific values be passed to `imageBuild()`
+The Docker `Image` class expects 3 or 4 arguments in its constructor call
 
-| Argument   | Type   | Required | Description                                       |
-|------------|--------|:--------:|---------------------------------------------------|
-| build_args | List   |     N    | `foo=bar` pairings for `docker build --build-arg` |
-| dockerfile | String |     Y    | Location of the Dockerfile to be built            |
-| image_name | String |     Y    | e.g. `invocaops/ruby_mysql`                       |
+| # | Type     | Required  | Description |
+|---|----------|:---------:|-------------|
+| 1 | Script   |     Y     | A reference to the Script object, always `this` when instantiated from the Jenkinsfile. |
+| 2 | String   |     Y     | The name of the image, including the Docker Hub organization.  i.e. `invocaops/ruby`. |
+| 3 | String[] |     Y     | An array of tags to apply to the image. |
+| 4 | String   |     N     | The directory that the Dockerfile is in.  Useful when multiple versions of the image need to be built.  Defaults to the directory the Jenkinsfile is in. |
```

Comment:
For my own understanding; These are numbered instead of named because the constructor is simply expecting the arguments in that order, correct?

Reply:
Correct.